main.go

```diff
@@ -83,34 +112,24 @@ func main() {
 		}
 	}
 
-	// Get hubconfig to talk to hub apiserver
-	if tool.Options.HubConfigFilePathName == "" {
-		found := false
-		tool.Options.HubConfigFilePathName, found = os.LookupEnv("HUB_CONFIG")
-		if found {
-			log.Info("Found ENV HUB_CONFIG, initializing using", "tool.Options.HubConfigFilePathName",
-				tool.Options.HubConfigFilePathName)
-		}
-	}
-
-	hubCfg, err := clientcmd.BuildConfigFromFlags("", tool.Options.HubConfigFilePathName)
+	managedClient, err := client.New(managedCfg, client.Options{Scheme: scheme})
 	if err != nil {
-		log.Error(err, "")
+		log.Error(err, "Failed to generate client to the managed cluster")
 		os.Exit(1)
 	}
 
-	ctx := context.TODO()
-	// Become the leader before proceeding
-	err = leader.Become(ctx, "policy-spec-sync-lock")
-	if err != nil {
-		log.Error(err, "")
-		os.Exit(1)
-	}
+	var kubeClient kubernetes.Interface = kubernetes.NewForConfigOrDie(managedCfg)
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartRecordingToSink(&corev1.EventSinkImpl{Interface: kubeClient.CoreV1().Events(namespace)})
+	managedRecorder := eventBroadcaster.NewRecorder(eventsScheme, v1.EventSource{Component: sync.ControllerName})
 
 	// Set default manager options
 	options := manager.Options{
```

Comment:
Could you also please add the ability to fallback to the legacy leader election as shown in this PR?
https://github.com/open-cluster-management/cert-policy-controller/pull/105

This is required for this controller to run on OCP 3.11.